html/user/openssl_sign_action.php

```diff
@@ -0,0 +1,148 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+//require_once('../inc/recaptchalib.php');
+
+// Needs to be defined, unknown use.
+check_get_args(array("tnow", "ttok"));
+
+// Check the user is online
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+// Optional: Recaptcha support
+//if ($recaptcha_private_key) {
```

Comment:
Since sites like gridcoin most likely directly link to this page, so it would be preferable to enable recaptcha by default if the project already uses it.

Reply:
I've enabled this by default, but when I had it enabled it threw some pretty ugly errors in black text at the top if I got it wrong. I've not got a live demo right now - broke it about a week after the conference, need to re-implement it to check this isn't still the case with the latest BOINC codebase 👍